mirrord-layer/tests/integration.rs

```diff
@@ -0,0 +1,252 @@
+    /// a mirror connection with the application.
+    /// Return the id of the new connection.
+    async fn send_new_connection(&mut self) -> u64 {
+        let new_connection_id = self.num_connections;
+        self.codec
+            .send(DaemonMessage::Tcp(DaemonTcp::NewConnection(
+                NewTcpConnection {
+                    connection_id: new_connection_id,
+                    address: "127.0.0.1".parse().unwrap(),
+                    destination_port: "80".parse().unwrap(),
+                    source_port: "31415".parse().unwrap(),
+                },
+            )))
+            .await
+            .unwrap();
+        self.num_connections += 1;
+        new_connection_id
+    }
+
+    async fn send_tcp_data(&mut self, message_data: &str, connection_id: u64) {
+        self.codec
+            .send(DaemonMessage::Tcp(DaemonTcp::Data(TcpData {
+                connection_id,
+                bytes: Vec::from(message_data),
+            })))
+            .await
+            .unwrap();
+    }
+
+    /// Tell the layer there is a new incoming connection, then send data "from that connection".
+    async fn send_connection_then_data(&mut self, message_data: &str) {
+        let new_connection_id = self.send_new_connection().await;
+        self.send_tcp_data(message_data, new_connection_id).await;
+    }
+}
+
+#[derive(Debug)]
+enum Application {
+    // TODO: add more applications.
+    PythonFlaskHTTP,
+}
+
+impl Application {
+    /// Run python with shell resolving to find the actual executable.
+    ///
+    /// This is to help tests that run python with mirrord work locally on systems with pyenv.
+    /// If we run `python3` on a system with pyenv the first executed is not python but bash. On mac
+    /// that prevents the layer from loading because of SIP.
+    async fn get_python3_executable() -> String {
+        let mut python = Command::new("python3")
+            .stdin(Stdio::piped())
+            .stdout(Stdio::piped())
+            .spawn()
+            .unwrap();
+        let child_stdin = python.stdin.as_mut().unwrap();
+        child_stdin
+            .write_all(b"import sys\nprint(sys.executable)")
+            .await
+            .unwrap();
+        let output = python.wait_with_output().await.unwrap();
+        String::from(String::from_utf8_lossy(&output.stdout).trim())
+    }
+
+    async fn get_executable(&self) -> String {
+        match self {
+            Application::PythonFlaskHTTP => Self::get_python3_executable().await,
+        }
+    }
+
+    fn get_args(&self) -> Vec<&str> {
+        match self {
+            Application::PythonFlaskHTTP => vec!["-u", "tests/apps/app_flask.py"],
+        }
+    }
+}
+
+/// Build the cdylib and return a path to it.
+#[fixture]
+#[once]
+fn dylib() -> PathBuf {
+    println!("Building project...");
+    let before_build = Instant::now();
+    let dylib_path = test_cdylib::build_current_project();
```

Comment:
Building can take quite some time on the CI (25 seconds in [a typical example](https://github.com/metalbear-co/mirrord/actions/runs/3188761789/jobs/5201818583#step:8:29) and 121 seconds in [this extreme case](https://github.com/metalbear-co/mirrord/actions/runs/3188136393/jobs/5200519924#step:8:600)). Should we maybe try to use an artefact of the `build_mirrord` job (and introduce a `build_mirrord_macos` job)? That would save 1 build (so typically 20~30 seconds) and would allow this test to have a tighter timeout without being flakey.

Reply:
sure